contracts/staking/MultiFeeDistribution.sol

```diff
@@ -349,16 +358,16 @@ contract MultiFeeDistribution is IMultiFeeDistribution, ReentrancyGuard, Ownable
         totalSupply = totalSupply.sub(adjustedAmount);
         stakingToken.safeTransfer(msg.sender, amount);
         if (penaltyAmount > 0) {
+            incentivesController.claim(address(this), new address[](0));
             _notifyReward(address(stakingToken), penaltyAmount);
         }
         emit Withdrawn(msg.sender, amount);
     }
 
-    // Claim all pending staking rewards
-    function getReward() public nonReentrant updateReward(msg.sender) {
-        uint256 length = rewardTokens.length;
+    function _getReward(address[] memory _rewardTokens) internal {
+        uint256 length = _rewardTokens.length;
         for (uint i; i < length; i++) {
-            address token = rewardTokens[i];
+            address token = _rewardTokens[i];
             uint256 reward = rewards[msg.sender][token];
             if (i > 0) {
```

Comment:
This check against `i > 0` no longer works as intended because you are not necessarily iterating over the reward tokens in order.

Reply:
I think this also requires a check that `token` is a valid reward token.